can/interfaces/vector/vxlapi.py

```diff
@@ -0,0 +1,859 @@
+
+
+# porthandle
+XL_INVALID_PORTHANDLE = (-1)
+XLportHandle = ctypes.c_long
+pXLportHandle = ctypes.POINTER(XLportHandle)
+
+
+# structures for xlGetDriverConfig
+class s_xl_channel_config(ctypes.Structure):
+    _pack_ = 1
+    _fields_ = [
+        ('name', ctypes.c_char * (XL_MAX_LENGTH + 1)),
+        ('hwType', ctypes.c_ubyte), ('hwIndex', ctypes.c_ubyte),
+        ('hwChannel', ctypes.c_ubyte), ('transceiverType', ctypes.c_ushort),
+        ('transceiverState', ctypes.c_ushort),
+        ('configError', ctypes.c_ushort), ('channelIndex', ctypes.c_ubyte),
+        ('channelMask', XLuint64), ('channelCapabilities', ctypes.c_uint),
+        ('channelBusCapabilities', ctypes.c_uint), ('isOnBus', ctypes.c_ubyte),
+        ('connectedBusType', ctypes.c_uint), ('busParams', XLbusParams),
+        ('_doNotUse', ctypes.c_uint), ('driverVersion', ctypes.c_uint),
+        ('interfaceVersion', ctypes.c_uint), ('raw_data', ctypes.c_uint * 10),
+        ('serialNumber', ctypes.c_uint), ('articleNumber', ctypes.c_uint),
+        ('transceiverName', ctypes.c_char * (XL_MAX_LENGTH + 1)),
+        ('specialCabFlags', ctypes.c_uint), ('dominantTimeout', ctypes.c_uint),
+        ('dominantRecessiveDelay', ctypes.c_ubyte),
+        ('recessiveDominantDelay', ctypes.c_ubyte), ('connectionInfo',
+                                                     ctypes.c_ubyte),
+        ('currentlyAvailableTimestamps', ctypes.c_ubyte), (
+            'minimalSupplyVoltage', ctypes.c_ushort), (
+                'maximalSupplyVoltage', ctypes.c_ushort), (
+                    'maximalBaudrate', ctypes.c_uint), ('fpgaCoreCapabilities',
+                                                        ctypes.c_ubyte),
+        ('specialDeviceStatus',
+         ctypes.c_ubyte), ('channelBusActiveCapabilities', ctypes.c_ushort), (
+             'breakOffset', ctypes.c_ushort),
+        ('delimiterOffset', ctypes.c_ushort), ('reserved', ctypes.c_uint * 3)
+    ]
+
+
+XLchannelConfig = s_xl_channel_config
+
+
+class s_xl_driver_config(ctypes.Structure):
+    _fields_ = [('dllVersion', ctypes.c_uint), ('channelCount', ctypes.c_uint),
+                ('reserved', ctypes.c_uint * 10),
+                ('channel', XLchannelConfig * XL_CONFIG_MAX_CHANNELS)]
+
+
+XLdriverConfig = s_xl_driver_config
+
+
+class VectorError(CanError):
+
+    def __init__(self, status):
+        self.status = status
+
+    def __str__(self):
+        return get_error_string(self.status)
+
+
+def check_status(result, function, arguments):
+    if result > 0:
+        raise VectorError(result)
+    return result
+
+
+# =============================================================================
+# Functions calls
+# ===============
+
+if _xlapi_dll is not None:
+    # xlOpenDriver();
+    xlOpenDriver = _xlapi_dll.xlOpenDriver
+    xlOpenDriver.argtypes = []
+    xlOpenDriver.restype = XLstatus
+    xlOpenDriver.errcheck = check_status
+
+
+    def open_driver():
```

Comment:
We could use the `ctypesutil.LibraryMixin` or try abstract this argtype/resulttype/status checking better.